### ISSUE TITLE
pdns_auth does not respect multiple include diretories

### ISSUE THREAD

**Issue by Cthullu** [PowerDNS/pdns#12339]
* Programm: Authorative
* Issue type: Bug report

## Short description

Setting multiple include-dir directives leads to the following errors:
```
/etc/powerdns/pdns.d, /etc/powerdns/pdns.sec.d is not accessible: No such file or directory
fatal error: /etc/powerdns/pdns.d, /etc/powerdns/pdns.sec.d is not accessible: No such file or directory
```

## Environment

* OS: Alpine Linux 3.17.0
* PowerDNS version used: 4.7.2
* Software source: System package manager

## Steps to reproduce:

1. Generate a pdns.conf file containing the following lines:
```
include-dir=/etc/powerdns/pdns.d
include-dir+=/etc/powerdns/pdns.sec.d
```
2. Start PowerDNS authorative server

## Expected behaviour

All files ending with *.conf in both directories are included.

## Actual behaviour

PowerDNS reads both directories as one.
```
/etc/powerdns/pdns.d, /etc/powerdns/pdns.sec.d is not accessible: No such file or directory
fatal error: /etc/powerdns/pdns.d, /etc/powerdns/pdns.sec.d is not accessible: No such file or directory
```

## Other information

This seems to be an issue within the following codeblock:
https://github.com/PowerDNS/pdns/blob/eaa5e59b8bbf40f1a6aeaeebc0ad79c7d92e955a/pdns/arguments.cc#L566

**Comment by Habbie** [comment]
Indeed, this is not currently supported. I marked this 'helpneeded', which means we agree this would be nice, and will accept good patches.

**Comment by Cthullu** [comment]
At [docs.powerdns.com](https://doc.powerdns.com/authoritative/settings.html#) this is the example for the `+=` feature:

> You can use += syntax to set some variables incrementally, but this requires you to have at least one non-incremental setting for the variable to act as base setting. This is mostly useful for [include-dir](https://doc.powerdns.com/authoritative/settings.html#setting-include-dir) directive.

Maybe the page should be adapted to avoid people trying to use it, until it is implemented.

**Comment by Habbie** [comment]
Ah, I see how that is confusing. The text means: you might like to expand a variable in several separate included files. So, have `forward-zones=` (for Recursor) in `recursor.conf`, then you can use `forward-zones+=` in multiple included files. Want to PR an improvement to that text?

**Comment by Cthullu** [comment]
I think this may confuse people (as it did with me). This is definitely not something a dedicated PR needs.
Having the feature for the include-dir directive would be nice though. From my side, this is a "low-priority nice to have".

**Comment by Ji993** [comment]
I know this is an old issue, but there is still no syntax available to include multiple directories with the YAML style config.

I'm getting the following error:
`May 25 16:22:08 msg="YAML config found, but error occurred processing it" error="recursor: duplicate field `include_dir` at line 44 column 3" subsystem="config" level="0" prio="Error" tid="0" ts="1748182928.980" configname="/etc/pdns-recursor/recursor.yml"`

**Comment by Habbie** [comment]
> configname="/etc/pdns-recursor

this ticket is about auth, not recursor.
